src/isEqual.test.ts

```diff
@@ -0,0 +1,47 @@
+import { isEqual } from './isEqual';
```

Comment:
Add tests that test the more interesting parts of your implementation too, specifically I'd like to see some more tests for arrays and objects and for narrowing of types.